### PR TITLE
[agnhost and other images] Use tonistiigi/binfmt instead of our multi-arch register.sh

### PR DESCRIPTION
**Pull request by dims**
Looking deeper into the logs there are a lot of errors like:
`script exited with error 1`

Initial reaction was that there was a problem with download, but it
looks like the script we use to register the qemu emulators may be at
fault, let's try this alternate mechanism.

Original issue : https://github.com/kubernetes/kubernetes/issues/101677
Revert the PR here : https://github.com/kubernetes/kubernetes/pull/101800 

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/images/image-util.sh [hunk 1]
```diff
@@ -165,7 +165,7 @@ build() {
             if [[ $(id -u) -ne 0 ]]; then
 	            sudo="sudo"
             fi
-            ${sudo} "${KUBE_ROOT}/third_party/multiarch/qemu-user-static/register/register.sh" --reset -p yes
+             ${sudo} docker run --rm --privileged tonistiigi/binfmt:latest --install all
```
Comment (hasheddan) [inline test/images/image-util.sh:168]: Does this intentionally have leading whitespace?

Comment (dims) [reply test/images/image-util.sh:168]: whoops! fixing

**Review by hasheddan — APPROVED** [review]
/lgtm

**Pull request merged**